sdcm/nemesis.py

```diff
@@ -434,7 +438,123 @@ def disrupt_stop_start_scylla_server(self):  # pylint: disable=invalid-name
+    def disrupt_start_stop_cleanup_compaction(self):
+        node = self.cluster.nodes[0]
+        compaction_ops = CompactionOps(self.cluster)
+        timeout = 120
+        trigger_func = partial(compaction_ops.trigger_cleanup_compaction)
+        watch_func = partial(compaction_ops.stop_on_user_compaction_logged,
+                             node=node,
+                             timeout=timeout,
+                             watch_for="Cleaning",
+                             stop_func=compaction_ops.stop_cleanup_compaction)
+        self.tester.wait_no_compactions_running()
+        ParallelObject(objects=[trigger_func, watch_func], timeout=timeout).call_objects()
+
+    def disrupt_start_stop_validation_compaction(self):
+        node = self.cluster.nodes[0]
+        compaction_ops = CompactionOps(self.cluster)
+        timeout = 120
+        trigger_func = partial(compaction_ops.trigger_validation_compaction)
+        watch_func = partial(compaction_ops.stop_on_user_compaction_logged,
+                             node=node,
+                             watch_for="Scrubbing in validate mode",
+                             timeout=timeout,
+                             stop_func=compaction_ops.stop_validation_compaction)
+        self.tester.wait_no_compactions_running()
+        ParallelObject(objects=[trigger_func, watch_func], timeout=timeout).call_objects()
+
+    def disrupt_start_stop_upgrade_compaction(self):
+        node = self.cluster.nodes[0]
+        compaction_ops = CompactionOps(self.cluster)
+        timeout = 300
+        upgraded_configuration_options = {"enable_sstables_mc_format": False,
+                                          "enable_sstables_md_format": True}
+
+        def _upgrade_sstables_format(node: BaseNode):
+            LOGGER.info("Upgrading sstables format...")
+            with node.remote_scylla_yaml() as scylla_yaml:
+                scylla_yaml.update(upgraded_configuration_options)
+
+        node.run_nodetool("flush")
+        node.stop_scylla()
+        _upgrade_sstables_format(node)
+        node.start_scylla()
+
+        trigger_func = partial(compaction_ops.trigger_upgrade_compaction)
+        watch_func = partial(compaction_ops.stop_on_user_compaction_logged,
+                             node=node,
+                             watch_for="Upgrade keyspace1.standard1",
+                             timeout=timeout,
+                             stop_func=compaction_ops.stop_upgrade_compaction)
+        self.tester.wait_no_compactions_running()
+        ParallelObject(objects=[trigger_func, watch_func], timeout=timeout).call_objects()
+
+    def disrupt_start_stop_reshape_compaction(self):
+        node = self.cluster.nodes[0]
+        compaction_ops = CompactionOps(self.cluster)
+        timeout = 300
+
+        def _trigger_reshape(node: BaseNode, tester: ClusterTester, keyspace: str = "keyspace1"):
+            twcs = {'class': 'TimeWindowCompactionStrategy', 'compaction_window_size': 1,
```

Comment:
better to revert the compaction strategy change

Reply:
As i moved this test to the test module (the Nemesis was crashing cassandra-stress runs), I skipped the revert. This test is now run as the last subTest in sequence, so reverting would not add anything.